src/dotnet-install.ps1

```diff
@@ -215,6 +226,54 @@ function Get-CLIArchitecture-From-Architecture([string]$Architecture) {
     }
 }
 
+
+function Get-NormalizedQuality([string]$Quality) {
+    Say-Invocation $MyInvocation
+
+    if ([string]::IsNullOrEmpty($Quality)) {
+        return ""
+    }
+
+    switch ($Quality.ToLower()) {
+        { ($_ -eq "daily") -or ($_ -eq "signed") -or ($_ -eq "validated") -or ($_ -eq "preview")} { return $Quality.ToLower() }
+        #ga quality is available without specifying quality, so normalizing it to empty
+        { $_ -eq "ga" } { return "" }
+        default { throw "'$Quality' is not a supported value for --quality option, supported values are: daily, signed, validated, preview, ga. If you think this is a bug, report it at https://github.com/dotnet/install-scripts/issues." }
```

Comment:
You have parameter validation on the top, you won't see that nice error from `default` if you specify an incorrect value. Which is a shame.

Reply:
would it be better to remove validation in this case?

Reply:
I would add ErrorMessage property on the ValidateSet: 


From https://adamtheautomator.com/powershell-validateset/#The_ErrorMessage_Property: Use the ErrorMessage property of the ValidateSet validation attribute to set a different error message, as shown in the example below. {0} is automatically replaced with the submitted value and {1} is automatically replaced with the list of permitted values.
```powershell
Param(
     [Parameter(Mandatory)]
     [ValidateSet("Mercury","Venus","Earth","Mars","Jupiter","Saturn","Uranus","Neptune","Pluto",ErrorMessage="Value '{0}' is invalid. Try one of: '{1}'")]
     $Planet
 )
```
Looking at it now, it will also be -Quality option, rather than --quality.

Reply:
`ErrorMessage` is only added in PowerShell 6.1. Should we use it anyway? 
@bozturkMSFT?

Reply:
Powershell version 6 and above is actually Powershell Core and needs to be installed manually. So `ErrorMessage` would only work on the machines of a tiny fraction of our customers.

Reply:
Ok, I removed ValidateSet, so the better error messages are used, I'm open to revert it back, though.